src/ripple/app/ledger/Ledger.cpp

```diff
@@ -1231,10 +992,11 @@ loadByIndex(std::uint32_t ledgerIndex, Application& app, bool acquire)
 {
     std::shared_ptr<Ledger> ledger;
     {
-        std::ostringstream s;
-        s << "WHERE LedgerSeq = " << ledgerIndex;
-        std::tie(ledger, std::ignore, std::ignore) =
-            loadLedgerHelper(s.str(), app, acquire);
+        std::optional<LedgerInfo> info =
+                app.getSQLInterface().loadLedgerInfoByIndex(
+                ledgerIndex);
+        if (info)
+            ledger = loadLedgerHelper(*info, app, acquire);
```

Comment:
What do you think of refactoring this as follows:
```

std::shared_ptr<Ledger>
loadByIndex(std::uint32_t ledgerIndex, Application& app, bool acquire)
{
    if (std::optional<LedgerInfo> info =
            app.getSQLInterface().loadLedgerInfoByIndex(ledgerIndex))
    {
        std::shared_ptr<Ledger> ledger = loadLedgerHelper(*info, app, acquire);
        finishLoadByIndexOrHash(ledger, app.config(), app.journal("Ledger"));
        return ledger;
    }
    return {};
}
```

Reply:
Done.